tee-worker/omni-executor/intent/executors/cross-chain/src/lib.rs

```diff
@@ -1110,7 +1111,8 @@ impl<BinanceClient: BinanceApi, SolanaClient: SolanaClientTrait>
 			.iter()
 			.map(|s| {
 				let hex_str = s.trim_start_matches("0x");
-				hex::decode(hex_str).expect("Invalid hex string")
+				let hex_decoded = hex::decode(hex_str).expect("Invalid hex string");
+				keccak_256(&hex_decoded).to_vec()
```

Comment:
hold on, in the `request_signatures` impl, there's already:

```
		let signature =
			self.request_signer.sign_prehashed(&keccak_256(&serialized_multi_sign_wallet)).0;
```

could you please verify if the hashing is still needed?

Reply:
the hashing is happening here only for signing, but the payload will still have the messages without hashing being done i.e. it is being passed as it is.

Reply:
Code ref 
https://github.com/litentry/heima/blob/4d2b4e415b33158f157178a7545c8fd0f0f8b4d5/tee-worker/omni-executor/pumpx/src/signer_client.rs#L237